src/server/DeviceServer.js

```diff
@@ -837,6 +850,28 @@ class DeviceServer {
       this._eventPublisher.publish(eventData);
     });
   }
+  publishIPCEvent = (
```

Comment:
i don't think that we need one more method for it, just add one more nullable argument in publish()